jiant/modules/sentence_encoder.py

```diff
@@ -11,13 +11,13 @@
 from allennlp.nn import InitializerApplicator, util
 from allennlp.modules import Highway, TimeDistributed
 
-from ..bert.utils import BertEmbedderModule
-from ..tasks.tasks import PairClassificationTask, PairRegressionTask
-from ..utils import utils
-from .simple_modules import NullPhraseLayer
-from .bilm_encoder import BiLMEncoder
-from .onlstm.ON_LSTM import ONLSTMStack
-from .prpn.PRPN import PRPN
+from jiant.pytorch_transformers_interface.modules import PytorchTransformersEmbedderModule
```

Comment:
There is a "global" bug when adding XLNet to sentence_encoder, the padding token of XLNet is not 0, so the sent_mask of sentence encoder is not computed correctly. i.e. be a full 1 matrix regardless of the actual length of the sentences.

Reply:
I checked this on demo with a print statement, and the masks all look right (and non-trivial). AllenNLP adds the padding, not XLNet, so it's set to zero. We then create the mask, and only after that, we rewrite the padding tokens to XLNet padding tokens.

Reply:
Well, my prints show the opposite result. I printed the sent_mask and batch['input1']['pytorch_transformers_wpm_pretokenized'] (I'll call this input1) in _single_sentence_forward (line 831). There are a lot of 5 in the right side of input1, 5 is also the _pad_id in XLNetEmbedderModule. And the sent_mask is all 1. Indeed the mask inside XLNetEmbedderModule.forward is correct, but that mask is only used in prepare_output. While this sent_mask, made by sent_encoder, is wrong, and it is going to make the Pooler in SingleClassifier work wrongly.

Reply:
Got it, thanks! I didn't understand that that was the line you were referring to. I'll take a look.

Reply:
Fixed. It was a subtle-ish reference-sharing issue—XLNet was modifying sent.